src/adrs/0129349218/README.md

```diff
@@ -0,0 +1,532 @@
+ * error messages.
+ *
+ * It is intended for a human audience. Simple programs display the message
+ * directly to the end user if they encounter an error condition they don't
+ * know how or don't care to handle. Sophisticated programs with more
+ * exhaustive error handling and proper internationalization are more likely
+ * to ignore the error message.
+ *
+ * Messages MAY use {placeholders} to reference values provided in the
+ * `info.metadata` field. For example, "The {account_id} is not found.".
+ */
+type MessageDescription = string;
+
+
+/**
+ * Source is the source of the error event. It contains information about the
+ * producer of the error event.
+ */
+type Source = {
+  /**
+   * Subject describes the subject of the error event in the context of the
+   * error event producer (identified by domain). It MAY be a JSON-POINTER, or
+   * an application-specific scheme to create unique identifiers for the
+   * subject.
+   */
+  subject?: string;
+}
+
+/**
+ * IndeterministicInfo contains information that is not deterministic and
+ * could change over time. Critical for debugging purposes, but not critical
+ * for the end-user, unless they give the information to the support team.
+ */
+type IndeterministicInfo = {
+  /**
+   * ID identifies the error. The combination of info + id is unique for each
+   * distinct error event. Consumers MAY assume that error events with identical
+   * source and id are duplicates.
+   */
+  id: string;
+
+  /**
+   * Time timestamp of when the occurrence happened.
+   */
+  time: Timestamp;
+}
+
+/**
+ * Domain identifies the context in which an error happened.
+ */
+type ErrorInfo = {
+  /**
+   * The reason of the error. This is a constant value that identifies the
+   * proximate cause of the error. Error reasons are unique within a particular
+   * domain of errors. This should be at most 63 characters and match a regular
+   * expression of `[A-Z][A-Z0-9_]+[A-Z0-9]`, which represents UPPER_SNAKE_CASE.
+   */
+  reason: string;
+
+  /**
+   * The logical grouping to which the "reason" belongs. The error domain is
+   * typically the registered service name of the tool or product that generates
+   * the error. Example: "pubsub.googleapis.com". If the error is generated by
+   * some common infrastructure, the error domain must be a globally unique value
+   * that identifies the infrastructure. For Google API infrastructure, the error
+   * domain is "googleapis.com".
+   */
+  domain: string;
+
+  /**
+   * Additional structured details about this error.
+   */
+  metadata: Record<string, string>;
+}
+
+type HelpLink = {
+  /**
+   * Describes what the link offers.
+   */
+  description: string;
+
+  /**
+   * The URL of the link.
+   */
+  url: string;
+}
+
+type Help = {
+  /**
+   * URL(s) pointing to additional information on handling the current error.
+   */
+  links: Array<HelpLink>;
+}
+
+/**
+ * Stacktrace contains the stack trace of the error.
```

Comment:
Interesting to pass the stack trace around. (Assume this is NOT for end user exposure)

A practice we've been experimenting with is wherever we create an error, we also log out through a common error reporting module. Our log contains a correlation ID that we also put on the error. 

This means we don't need to put things like stacktraces on the error, as the error will tell us where to find the stacktrace. Curious about your thoughts on this!

Reply:
@acco I added the "visibility" dimension as well, after @dallashall feedback